pkg/lib/crd/storage.go

```diff
@@ -0,0 +1,66 @@
+package crd
+
+import (
+	"fmt"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// SafeStorageVersionUpgrade determines whether the new CRD spec includes all the storage versions of the existing on-cluster CRD.
+// For each stored version in the status of the CRD on the cluster (there will be at least one) - each version must exist in the spec of the new CRD that is being installed.
+// See https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definition-versioning/#upgrade-existing-objects-to-a-new-stored-version.
+func SafeStorageVersionUpgrade(existingCRD runtime.Object, newCRD runtime.Object) error {
+	newSpecVersions, existingStatusVersions := getStoredVersions(existingCRD, newCRD)
+	if newSpecVersions == nil {
+		return fmt.Errorf("could not find a stored version in the spec of the new CRD: found %#v", newSpecVersions)
+	}
+	if existingStatusVersions == nil {
```

Comment:
I think this should return nil - if there are no existing stored versions, we don't care what the new versions are, right?

Reply:
In reality, this never happens given at least one version must be storage version. If user attempts to remove the only storage version in the status, I'm pretty sure it will error out on kubeclt.

Reply:
yes, but is this an error on the kube side of things - every CRD on the cluster has at-least one storage version in its status right? I agree its not an OLM issue exactly. 

I guess returning early here is safe